services/noteservice/noteservice.go

```diff
@@ -0,0 +1,48 @@
+/*
+Package noteservice handles interactions with database layer.
+*/
+package noteservice
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/atmiguel/cerealnotes/databaseutil"
+	"github.com/atmiguel/cerealnotes/models"
+)
+
+func StoreNewNote(
+	note *models.Note,
+) (models.NoteId, error) {
+
+	id, err := databaseutil.InsertNewNote(int64(note.AuthorId), note.Content, note.CreationTime)
+	if err != nil {
+		return models.NoteId(0), err
+	}
+
+	return models.NoteId(id), nil
+}
+
+func StoreNewNoteCategoryRelationship(
+	noteId models.NoteId,
+	category models.Category,
+) error {
+	if err := databaseutil.InsertNoteCategoryRelationship(int64(noteId), category.String()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type NoteMap map[models.NoteId]*models.Note
+
+func (noteMap NoteMap) ToJson() ([]byte, error) {
+	// json doesn't support int indexed maps
+	datas := make(map[string]models.Note, len(noteMap))
+
+	for id, note := range noteMap {
+		datas[fmt.Sprint(id)] = *note
```

Comment:
this is likely another reason people hate passing maps through http, because json is so limited. Another option here (that i expect you may dislike), is that in postgres, we could have the concept of a public and a private id. The current id would become the private id (meaning it would mostly stay in the database layer), and we'd assign a UUID as the public id for each note, upon creation. Then in our codebase, noteId would cease to be an integer, and would be a string instead. Thoughts?

Reply:
Both are valid options. I personally don't think the benefits of creating a uuid column is worth the change at least not right now.